tst/testinstall/sha256.tst

```diff
@@ -0,0 +1,28 @@
+#
+gap> START_TEST("sha256.tst");
+
+#
+# test input validation for the kernel functions
+#
+gap> state := GAP_SHA256_INIT();
+<object>
+
+#
+gap> GAP_SHA256_UPDATE(fail, fail);
+Error, GAP_SHA256_UPDATE: <state> must be a SHA256 state (not the value 'fail'\
+)
+gap> GAP_SHA256_UPDATE(state, fail);
+Error, GAP_SHA256_UPDATE: <bytes> must be a string (not the value 'fail')
+gap> GAP_SHA256_HMAC(fail, fail);
+Error, GAP_SHA256_HMAC: <key> must be a string (not the value 'fail')
+gap> GAP_SHA256_HMAC("", fail);
+Error, GAP_SHA256_HMAC: <text> must be a string (not the value 'fail')
+
+#
+gap> HexSHA256String("abcd");
+"88d4266fd4e6338d13b845fcf289579d209c897823b9217da3e161936f031589"
+gap> HexSHA256String("abcd\n");
+"fc4b5fd6816f75a7c81fc8eaa9499d6a299bd803397166e8c4cf9280b801d62c"
```

Comment:
This is very low level, but maybe add:

`HexSHA256String(['a', 'b', 'c', 'd']);` (same as string).
`HexSHA256String("");` (should be 0xe3b0c44298fc1c149afbf4c8996fb92427ae41e4649b934ca495991b7852b855 according to wikipedia?)
`HexSHA256String("abcd\r\n"); (just to check nothing stupid happens on windows, can't imagine how it would).

Reply:
Good ideas. Also, the documentation needs to be hooked up so it actually appears in the manual ;-), and a few other things. I'll work on it later